### PR TITLE
Allow sorting consumer groups by topic num

### PR DESCRIPTION
**Pull request by nisanohana3**
**What changes did you make?** (Give an overview)

* Added an additional value through the enum of ConsumerGroupOrdering
* Enabled sorting in the react consumer groups page (by topic num)
* Moved the topic num calculation logic to InternalConsumerGroup as part of its creation
* Added the topic num case - sorting consumerGroupDescription according to comparator with topic num logic

**Screenshot of the feature**
image

**Is there anything you'd like reviewers to focus on?**
The new methods that was added to internalConsumerGroup and consumerGroupService as part the feature

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually  - I opened the consumer group page and clicked on the topic num colum, ensured it was sorting properly
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

**A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalConsumerGroup.java [hunk 1]
```diff
@@ -80,4 +72,29 @@ private static Long calculateMessagesBehind(Map<TopicPartition, Long> offsets, M
     return messagesBehind;
   }
 
+  private static Integer calculateTopicNum(Map<TopicPartition, Long> offsets, Collection<InternalMember> members) {
+
+    return Stream.concat(
+        offsets.keySet().stream().map(TopicPartition::topic),
+        members.stream()
+            .flatMap(m -> m.getAssignment().stream().map(TopicPartition::topic))
+    ).collect(Collectors.toSet()).size();
```
Comment (iliax) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalConsumerGroup.java:81]: Please use `.distinct().count()` instead

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java [hunk 2]
```diff
@@ -209,6 +209,27 @@ private Mono<List<ConsumerGroupDescription>> describeConsumerGroups(ReactiveAdmi
...
+  private Mono<List<ConsumerGroupDescription>> loadDescriptionsByInternalConsumerGroups(ReactiveAdminClient ac,
+                                                                                  List<ConsumerGroupListing> groups,
+                                                                                  Comparator<GroupWithDescr> comparator,
+                                                                                  int pageNum,
+                                                                                  int perPage,
+                                                                                  SortOrderDTO sortOrderDto) {
+    var groupNames = groups.stream().map(ConsumerGroupListing::groupId).toList();
+
+    return ac.describeConsumerGroups(groupNames)
+    .flatMap(descriptionsMap -> {
```
Comment (iliax) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java:222]: please format code using `/etc/checkstyle.xml`

**Review by iliax — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalConsumerGroup.java (hunk 1 above)
Comment (nisanohana3) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalConsumerGroup.java]: Done. 
In order to have compatibility to the topic field type in consumerGroupDTO I casted the resulted long to Integer. is it fine ?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java (hunk 2 above)
Comment (nisanohana3) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java:222]: The build finishes successfully without any errors or warnings (from the checkstyle plugin) regarding style issues here. 
What exactly is the issue ? what should I do ?

**Comment by nisanohana3** [comment]
Is there anything else I can do on my side to make progress ?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalConsumerGroup.java [hunk 3]
```diff
@@ -80,4 +72,31 @@ private static Long calculateMessagesBehind(Map<TopicPartition, Long> offsets, M
...
+
+    long topicNum = Stream.concat(
+        offsets.keySet().stream().map(TopicPartition::topic),
+        members.stream()
+            .flatMap(m -> m.getAssignment().stream().map(TopicPartition::topic))
+    ).distinct().count();
+
+    return Integer.valueOf((int) topicNum);
+
+  }
```
[the remark this answers is not in the document]
Comment (Haarolean) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/model/InternalConsumerGroup.java:85]: Fixed

**Review by Haarolean — APPROVED** [review]

**Review by iliax — APPROVED** [review]

**Comment by Haarolean** [comment]
@nisanohana3 thank you

**Pull request merged**